cql3/restrictions/statement_restrictions.cc

```diff
@@ -1021,5 +1022,42 @@ ::shared_ptr<single_column_restriction> single_column_restriction::LIKE::apply_t
     return r;
 }
 
+namespace wip {
+
+::shared_ptr<expression> make_conjunction(::shared_ptr<expression> a, ::shared_ptr<expression> b) {
+    if (!a) {
+        return b;
+    }
+    if (!b) {
+        return a;
+    }
+    return std::visit(overloaded_functor{
+            [&] (const conjunction& a_conj) {
+                auto operands = a_conj.children;
+                std::visit(overloaded_functor{
+                        [&] (const conjunction& b_conj) {
+                            operands.insert(operands.end(), b_conj.children.cbegin(), b_conj.children.cend());
+                        },
+                        [&] (const binary_operator&) {
```

Comment:
This will just get bloated as we add more and more members to the variant. I think we can use const auto& here as a catch-all.

Reply:
Obsolete after a rewrite.